src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -144,7 +145,8 @@ def validate_client_parameters(cmd, namespace):
                            ' ,'.join(account_key_args))
         return
 
-    if not n.connection_string:
+    # When there is no input for credential, we will read environment variable
+    if not n.connection_string and not n.account_key:
```

Comment:
```suggestion
    if not n.connection_string and not n.account_key and not n.sas_token:
```
If `sas_token` is given, do we need to read from environment variable?

Reply:
goo point. Fixed in two places.